src/test/unit/profile-data.test.js

```diff
@@ -637,6 +643,27 @@ describe('upgrades', function() {
       serializedOldProcessedProfile6
     );
     compareProcessedProfiles(upgradedProfile6, afterUpgradeReference);
+
+    const serializedOldProcessedProfile7 = require('../fixtures/upgrades/processed-7.json');
+    const upgradedProfile7 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile7
+    );
+    compareProcessedProfiles(upgradedProfile7, afterUpgradeReference);
+
+    const serializedOldProcessedProfile7a = require('../fixtures/upgrades/processed-7a.json');
+    const afterUpgradeReference8a = unserializeProfileOfArbitraryFormat(
+      require('../fixtures/upgrades/processed-8a.json')
+    );
+    const upgradedProfile7a = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile7a
+    );
+    compareProcessedProfiles(upgradedProfile7a, afterUpgradeReference8a);
+
```

Comment:
nit: please add a comment like "This last test is to make sure we properly upgrade the json file".